tests/foreman/api/test_media.py

```diff
@@ -16,60 +16,73 @@
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.decorators import tier1
 from robottelo.decorators import tier2
 from robottelo.decorators import upgrade
-from robottelo.test import APITestCase
 
 
-class MediaTestCase(APITestCase):
+class TestMedia:
     """Tests for ``api/v2/media``."""
 
-    @classmethod
-    def setUpClass(cls):
-        """Create an organization which can be re-used in tests."""
-        super(MediaTestCase, cls).setUpClass()
-        cls.org = entities.Organization().create()
+    @pytest.fixture(scope='class')
+    def class_media(self, module_org):
+        return entities.Media(organization=[module_org]).create()
 
     @tier1
-    def test_positive_create_with_name(self):
-        """Create media with valid name only
+    @upgrade
+    @pytest.mark.parametrize(
+        'name, new_name',
+        **parametrized(list(zip(valid_data_list().values(), valid_data_list().values())))
+    )
+    def test_positive_crud_with_name(self, module_org, name, new_name):
+        """Create, update, delete media with valid name only
+
+        :id: b07a4549-7dd5-4b36-a1b4-9f8d48ddfcb5
 
-        :id: 892b44d5-0f11-4e9d-8ee9-fd5abe0b0a9b
+        :parametrized: yes
 
         :expectedresults: Media entity is created and has proper name
 
         :CaseImportance: Critical
         """
-        for name in valid_data_list().values():
-            with self.subTest(name):
-                media = entities.Media(organization=[self.org], name=name).create()
-                self.assertEqual(media.name, name)
+        media = entities.Media(organization=[module_org], name=name).create()
+        assert media.name == name
+        media = entities.Media(id=media.id, name=new_name).update(['name'])
+        assert media.name == new_name
+        media.delete()
+        with pytest.raises(HTTPError):
+            media.read()
 
     @tier1
-    def test_positive_create_with_os_family(self):
-        """Create media with every OS family possible
+    @pytest.mark.parametrize('os_family', **parametrized(OPERATING_SYSTEMS))
+    def test_positive_create_update_with_os_family(self, module_org, os_family):
+        """Create and update media with every OS family possible
 
-        :id: 7885e205-6189-4e71-a6ee-e5ddb077ecee
+        :id: d02404f0-b2ad-412c-b1cd-0548254f7c88
+
+        :parametrized: yes
 
         :expectedresults: Media entity is created and has proper OS family
             assigned
         """
-        for os_family in OPERATING_SYSTEMS:
-            with self.subTest(os_family):
-                media = entities.Media(organization=[self.org], os_family=os_family).create()
-                self.assertEqual(media.os_family, os_family)
+        media = entities.Media(organization=[module_org], os_family=os_family).create()
+        assert media.os_family == os_family
+        new_os_family = OPERATING_SYSTEMS[random.randint(1, len(OPERATING_SYSTEMS) - 1)]
```

Comment:
`random.choice()` returns random element from iterable:
```suggestion
        new_os_family = random.choice(OPERATING_SYSTEMS)
```